### PR TITLE
Adds an EMP protection component, allowing ANYTHING to be protected from EMP

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
The component is initialized with any combination of three flags, EMP_PROTECT_SELF protects against effects that target the object itself, EMP_PROTECT_WIRES protects against wires being messed with similar to the NO_EMP_WIRES_1 flag which this PR removes, EMP_PROTECT_CONTENTS protects against things that are inside the object like organs, internal "non-virtual" power cells and the like

This enables (but doesn't introduce) new mechanics such as adding EMP shielding at runtime to any atom, or taking it away.

:cl: Naksu
code: Added EMP protection component, replaced various bespoke ways things handled EMP proofness
/:cl:

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
should probably just refactor empulse and get rid of all this other crap instead, this is what you get when you develop between videogame rounds

File: code/game/atoms.dm [hunk 1]
```diff
@@ -221,9 +221,10 @@
 	return
 
 /atom/proc/emp_act(severity)
-	SendSignal(COMSIG_ATOM_EMP_ACT, severity)
-	if(istype(wires) && !(flags_1 & NO_EMP_WIRES_1))
+	var/protection = SendSignal(COMSIG_ATOM_EMP_ACT, severity)
+	if(!(protection & EMP_PROTECT_WIRES) && istype(wires))
```
Comment (optimumtact) [inline code/game/atoms.dm]: this behaviour should really live in the components dude

Comment (optimumtact) [reply code/game/atoms.dm]: I guess that's a bigger overhaul for another time

**Comment by adamkad1** [comment]
make a op borg

**Pull request merged**